api/src/task/notifications/generate_notifications.py

```diff
@@ -103,32 +98,42 @@ def _collect_search_notifications(self) -> None:
     def _send_notifications(self) -> None:
         """Send collected notifications to users"""
         for user_id, container in self.user_notification_map.items():
-            if not container.updated_opportunity_ids and not container.updated_searches:
+            if not container.saved_opportunities and not container.updated_searches:
                 continue
 
             # TODO: Implement actual notification sending in future ticket
             logger.info(
                 "Would send notification to user",
                 extra={
                     "user_id": user_id,
-                    "opportunity_count": len(container.updated_opportunity_ids),
+                    "opportunity_count": len(container.saved_opportunities),
                     "search_count": len(container.updated_searches),
                 },
             )
 
+            # Create notification log entry
+            # TODO: Use enum for notification reason?
+            notification_log = UserNotificationLog(
+                user_id=user_id,
+                notification_reason="opportunity_updates",
+                notification_sent=True,
+            )
```

Comment:
For the notification reason, I wasn't sure how notifications might evolve over time so left it pretty freeform. For this PR and the later one to add notifications for search, I'd probably just suggest adding some constants at the top of the file like:
```
class NotificationConstants:
     OPPORTUNITY_UPDATES = "opportunity_updates"
```

At least organizes it slightly kinda like an enum